lib/views/drawer/my_drawer.dart

```diff
@@ -0,0 +1,71 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:game_lib_app/blocs_and_cubits/drawer_cubit.dart';
+import 'package:game_lib_app/blocs_and_cubits/signin/signin_cubit.dart';
+import 'package:game_lib_app/blocs_and_cubits/signup/signup_cubit.dart';
+import 'package:game_lib_app/repositories/fb_auth_repository.dart';
+import 'package:game_lib_app/views/drawer/login_view.dart';
+import 'package:game_lib_app/views/drawer/main_drawer_view.dart';
+import 'package:game_lib_app/views/drawer/register_view.dart';
+
+class MyDrawer extends StatelessWidget {
+  MyDrawer({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    Widget content = const SizedBox();
+    switch (context.watch<DrawerCubit>().state) {
+      case DrawerState.initial:
+        {
+          content = const MainDrawerView();
+        }
+        break;
+      case DrawerState.login:
+        {
+          content = const LoginView();
+        }
+        break;
+      case DrawerState.register:
+        {
+          content = const RegisterView();
+        }
+        break;
+    }
```

Comment:
Why you don't want to bloc builder instead of that switch? I'll go with bloc builder.

Reply:
i kind don't know how to do this logic without the switch, even when using blocBuilder

Reply:
in return of BlocBuilder: 

if (state == DrawerState.initial) return MainDrawerView();
if(state== DrawerState.login) return LoginView();
.....